py/tests/tests_unit_conf.py

```diff
@@ -0,0 +1,52 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Unit tests for PySparkling H2O Configuration
+"""
+
+import unittest
+from pysparkling.context import H2OContext
+from pysparkling.conf import H2OConf
+from pyspark import SparkContext, SparkConf
+from pyspark.sql import SparkSession
+from pyspark.sql import Row
+
+import h2o
+import test_utils
+
+class H2OConfTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls._spark = SparkSession.builder.config(conf = test_utils.get_default_spark_conf().set("spark.ext.h2o.cloud.name", "test-cloud")).getOrCreate()
+        test_utils.set_up_class(cls)
+        h2o_conf = H2OConf(cls._spark).set_num_of_external_h2o_nodes(2)
+        cls._hc = H2OContext.getOrCreate(cls._spark, h2o_conf)
+
+    @classmethod
+    def tearDownClass(cls):
+        h2o.cluster().shutdown()
+        test_utils.tear_down_class(cls)
+
+    # test passing h2o_conf to H2OContext
+    def test_h2o_conf(self):
+        self.assertEquals(self._hc.get_conf().cloud_name(),"test-cloud", "Configuration property cloud_name should match")
```

Comment:
is it possible to test the rest properties of config? Futhermore, the test should fail if we add a new config property.

Reply:
I created these 2 PRs ( just to keep this one simple ) https://0xdata.atlassian.net/browse/SW-578 - for testing all properties

https://0xdata.atlassian.net/browse/SW-579 - for testing consistency of properties.

We might actually automatically generate the python configuration based on the Scala one.